File: backend/src/main/java/travelplanner/dto/trip/TripResponse.java
package travelplanner.dto.trip;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public record TripResponse(
        UUID tripId,
        String title,
        String destination,
        LocalDate startDate,
        LocalDate endDate,
        String description,
        BigDecimal budget,
        String currency,
        String coverUrl,
        Boolean isPublic,
        LocalDateTime createdAt
) {}

File: frontend/src/types/TripResponse.ts
export interface TripResponse {
  tripId: string;
  title: string;
  destination: string;
  startDate: string;
  endDate: string;
  description: string | null;
  budget: number | null;
  currency: string | null;
  coverUrl: string | null;
  isPublic: boolean;
  createdAt: string;
}